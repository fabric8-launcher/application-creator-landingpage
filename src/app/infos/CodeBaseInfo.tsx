import {Grid, GridItem} from '@patternfly/react-core';
import * as React from 'react';
import {ExternalLink} from '../../shared/components/ExternalLink';
import InfoCard from '../components/InfoCard';
import {CodeIcon} from '@patternfly/react-icons';
import ShellCommand from '../../shared/components/ShellCommand';

interface CodeBaseInfoProps {
  runtime: {
    name: string;
    description: string;
    icon: string;
    metadata: {
      language: string;
    }
  };
  baseImage: string;
  sourceRepository?: {
    url: string;
    provider: string;
  };
}

export function CodeBaseInfo(props: CodeBaseInfoProps) {
  return (
    <InfoCard name="codebase">
      <InfoCard.Title><CodeIcon className="with-text"/> {props.runtime.name} - {props.runtime.metadata.language} - Codebase</InfoCard.Title>
      <InfoCard.Body>
        <Grid>
          <GridItem span={3}>
            <img src={props.runtime.icon}/>
          </GridItem>
          <GridItem span={9}>
            <h1>{props.runtime.name} - {props.runtime.metadata.language}</h1>
            <p>Your codebase has been customized to hook into the services selected during generation.
              All capabilities are functional end-to-end for you to explore.</p>
          </GridItem>
          <InfoCard.Separator/>

          {props.sourceRepository && (
            <React.Fragment>
              <GridItem span={3}>Git URL</GridItem>
              <GridItem span={9}>
                <ExternalLink href={props.sourceRepository.url}>{props.sourceRepository.url}</ExternalLink>
                <br/>
                You may use the git command-line client (or other preferred tooling) to clone this repository locally.
                <ShellCommand command={`git clone ${props.sourceRepository.url}`}/>
              </GridItem>
              <InfoCard.Separator/>
            </React.Fragment>
          )}

          <GridItem span={3}>Base Image</GridItem><GridItem span={9}>{props.baseImage}</GridItem>
        </Grid>
      </InfoCard.Body>
    </InfoCard>
  );
}