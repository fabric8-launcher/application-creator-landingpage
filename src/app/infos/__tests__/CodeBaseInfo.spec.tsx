import * as React from 'react';
import {shallow} from 'enzyme';
import { CodeBaseInfo } from '../CodeBaseInfo';

const runtime = {
  name: 'Vert.x',
  description: `A
  tool-kit for building reactive applications on the
  JVM.`,
  icon: 'icon',
  metadata: {
    language: 'java',
  },
};

const sourceRepository = {
  url: 'https://github.com/fabric8-launcher/launcher-creator-welcome-app.git',
  provider: 'GitHub',
};

describe('<CodeBaseInfo />', () => {
  it('check that render is correct', () => {
    const component = shallow(<CodeBaseInfo baseImage="group/imagename" runtime={runtime} />);
    expect(component).toMatchSnapshot();
  });

  it('check that render is correct with repositoryUrl', () => {
    const component = shallow(<CodeBaseInfo baseImage="group/imagename" runtime={runtime} sourceRepository={sourceRepository} />);
    expect(component).toMatchSnapshot();
  });

});

