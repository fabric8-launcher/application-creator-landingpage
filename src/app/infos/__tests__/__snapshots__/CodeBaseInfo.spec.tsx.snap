// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CodeBaseInfo /> check that render is correct 1`] = `
<InfoCard
  name="codebase"
>
  <Component>
    <CodeIcon
      className="with-text"
      color="currentColor"
      size="sm"
      title={null}
    />
     
    Vert.x
     - 
    java
     - Codebase
  </Component>
  <Component>
    <Grid
      className=""
      gutter={null}
      lg={null}
      md={null}
      sm={null}
      span={null}
      xl={null}
    >
      <GridItem
        className=""
        lg={null}
        lgOffset={null}
        lgRowSpan={null}
        md={null}
        mdOffset={null}
        mdRowSpan={null}
        offset={null}
        rowSpan={null}
        sm={null}
        smOffset={null}
        smRowSpan={null}
        span={3}
        xl={null}
        xlOffset={null}
        xlRowSpan={null}
      >
        <img
          src="icon"
        />
      </GridItem>
      <GridItem
        className=""
        lg={null}
        lgOffset={null}
        lgRowSpan={null}
        md={null}
        mdOffset={null}
        mdRowSpan={null}
        offset={null}
        rowSpan={null}
        sm={null}
        smOffset={null}
        smRowSpan={null}
        span={9}
        xl={null}
        xlOffset={null}
        xlRowSpan={null}
      >
        <h1>
          Vert.x
           - 
          java
        </h1>
        <p>
          Your codebase has been customized to hook into the services selected during generation. All capabilities are functional end-to-end for you to explore.
        </p>
      </GridItem>
      <Component />
      <GridItem
        className=""
        lg={null}
        lgOffset={null}
        lgRowSpan={null}
        md={null}
        mdOffset={null}
        mdRowSpan={null}
        offset={null}
        rowSpan={null}
        sm={null}
        smOffset={null}
        smRowSpan={null}
        span={3}
        xl={null}
        xlOffset={null}
        xlRowSpan={null}
      >
        Base Image
      </GridItem>
      <GridItem
        className=""
        lg={null}
        lgOffset={null}
        lgRowSpan={null}
        md={null}
        mdOffset={null}
        mdRowSpan={null}
        offset={null}
        rowSpan={null}
        sm={null}
        smOffset={null}
        smRowSpan={null}
        span={9}
        xl={null}
        xlOffset={null}
        xlRowSpan={null}
      >
        group/imagename
      </GridItem>
    </Grid>
  </Component>
</InfoCard>
`;

exports[`<CodeBaseInfo /> check that render is correct with repositoryUrl 1`] = `
<InfoCard
  name="codebase"
>
  <Component>
    <CodeIcon
      className="with-text"
      color="currentColor"
      size="sm"
      title={null}
    />
     
    Vert.x
     - 
    java
     - Codebase
  </Component>
  <Component>
    <Grid
      className=""
      gutter={null}
      lg={null}
      md={null}
      sm={null}
      span={null}
      xl={null}
    >
      <GridItem
        className=""
        lg={null}
        lgOffset={null}
        lgRowSpan={null}
        md={null}
        mdOffset={null}
        mdRowSpan={null}
        offset={null}
        rowSpan={null}
        sm={null}
        smOffset={null}
        smRowSpan={null}
        span={3}
        xl={null}
        xlOffset={null}
        xlRowSpan={null}
      >
        <img
          src="icon"
        />
      </GridItem>
      <GridItem
        className=""
        lg={null}
        lgOffset={null}
        lgRowSpan={null}
        md={null}
        mdOffset={null}
        mdRowSpan={null}
        offset={null}
        rowSpan={null}
        sm={null}
        smOffset={null}
        smRowSpan={null}
        span={9}
        xl={null}
        xlOffset={null}
        xlRowSpan={null}
      >
        <h1>
          Vert.x
           - 
          java
        </h1>
        <p>
          Your codebase has been customized to hook into the services selected during generation. All capabilities are functional end-to-end for you to explore.
        </p>
      </GridItem>
      <Component />
      <GridItem
        className=""
        lg={null}
        lgOffset={null}
        lgRowSpan={null}
        md={null}
        mdOffset={null}
        mdRowSpan={null}
        offset={null}
        rowSpan={null}
        sm={null}
        smOffset={null}
        smRowSpan={null}
        span={3}
        xl={null}
        xlOffset={null}
        xlRowSpan={null}
      >
        Git URL
      </GridItem>
      <GridItem
        className=""
        lg={null}
        lgOffset={null}
        lgRowSpan={null}
        md={null}
        mdOffset={null}
        mdRowSpan={null}
        offset={null}
        rowSpan={null}
        sm={null}
        smOffset={null}
        smRowSpan={null}
        span={9}
        xl={null}
        xlOffset={null}
        xlRowSpan={null}
      >
        <ExternalLink
          href="https://github.com/fabric8-launcher/launcher-creator-welcome-app.git"
        >
          https://github.com/fabric8-launcher/launcher-creator-welcome-app.git
        </ExternalLink>
        <br />
        You may use the git command-line client (or other preferred tooling) to clone this repository locally.
        <ShellCommand
          command="git clone https://github.com/fabric8-launcher/launcher-creator-welcome-app.git"
        />
      </GridItem>
      <Component />
      <GridItem
        className=""
        lg={null}
        lgOffset={null}
        lgRowSpan={null}
        md={null}
        mdOffset={null}
        mdRowSpan={null}
        offset={null}
        rowSpan={null}
        sm={null}
        smOffset={null}
        smRowSpan={null}
        span={3}
        xl={null}
        xlOffset={null}
        xlRowSpan={null}
      >
        Base Image
      </GridItem>
      <GridItem
        className=""
        lg={null}
        lgOffset={null}
        lgRowSpan={null}
        md={null}
        mdOffset={null}
        mdRowSpan={null}
        offset={null}
        rowSpan={null}
        sm={null}
        smOffset={null}
        smRowSpan={null}
        span={9}
        xl={null}
        xlOffset={null}
        xlRowSpan={null}
      >
        group/imagename
      </GridItem>
    </Grid>
  </Component>
</InfoCard>
`;
